### ISSUE TITLE
this.url for non-image file types

### ISSUE THREAD

**Issue by clarkritchie** [CollectionFS/Meteor-CollectionFS#614]
Surely I am missing something obvious here, but I can't seem to get ```this.url``` to work for non-image file types -- PDFs, etc.  ```this.url``` works fine for files that are images.

For non-images, here is my collection:
```
Files = new FS.Collection('files', {
    stores: [
        new FS.Store.FileSystem("files")
    ],
    filter: {
        allow: {
            extensions: ['pdf','doc','docx','xls','xlsx','zip']
        },
        onInvalid: function (message) {
            if (Meteor.isClient) {
                // my custom error handling here
            }
        }
    }
});
```

I am able to insert files (PDFs, etc.) OK.

When my page loads, I see files in the ```files``` data context, e.g.

```
Template.myAwesomeTemplate.rendered = function() {
    console.log(this.data.files.fetch()); // w00t, I see my files!
}
```

I can print each file's name, but I can't generate a clickable link.

```

		<dt>Files</dt>
		{{#each files}}
			<dd>
				<a href="{{this.url}}">{{this.original.name}}</a>
			</dd>
		{{/each}}
```

Virtually the same code (I have a different collection for images) works for PNGs, JPGs, etc.

![image](https://cloud.githubusercontent.com/assets/385550/6762596/c4fe102e-cf20-11e4-9aa6-d206112e87bd.png)

What am I missing?  I feel I've RTFM.

**Comment by raix** [comment]
I think this issue relates to the issue about partial download (this is used with pdf) #495

**Comment by aldeed** [comment]
@clarkritchie, in the generated HTML, is there a url in the `href`? If the file data wasn't properly uploaded, then the url helper might not be returning anything.

**Comment by clarkritchie** [comment]
@aldeed No, the href is blank.  

Here's the insert code, am I missing something?  ```addFileMetaData``` just tags the file with the userId of the person who uploaded it and their organization.  And in this example, ```otherOptions``` just adds one  more tag, the siteId.

```
'click #uploadFiles': function(e) {
        e.preventDefault();
        var uploadFiles = document.getElementById('files') || null;
        if (uploadFiles) {
            for (var x = 0; x < uploadFiles.files.length; x++) {
                var file = uploadFiles.files[x];
                Files.insert(addFileMetaData(file, {siteId: $('#siteId').val() }), function(err, fileObj) {
                    if (err) {
                        throwError(err.reason, DANGER, err.error, err.details);
                    } else {
                        // optionally do something with fileObj._id
                        // reset the control
                        $('#files').val('');
                    };
                });
            }
        }
    },
```

```
addFileMetaData = function(file, otherOptions) {
    if (file) {
        var fileWithMetaData = new FS.File(file);
        fileWithMetaData.metadata = {
            serviceProviderId: Meteor.user().serviceProviderId,
            createdBy: Meteor.user()._id
        };
        _.each(otherOptions, function(value, key){
            fileWithMetaData.metadata[key] = value;
        });

        return fileWithMetaData;
    }
}
```

**Comment by aldeed** [comment]
I don't see any issues. Are you doing `FS.debug = true` in your code, and is it logging anything useful about the failed upload?

**Comment by clarkritchie** [comment]
Ah, spaces -- I gather there's no support for having spaces in filenames?  When I renamed my file from **Large PDF.pdf** to **Large_PDF.pdf** the helper worked OK and I could click to view my file.

What's the recommendation on how to handle spaces, automatically convert them to underscores or something else?

**Comment by nooitaf** [comment]
Spaces shouldn't be a problem. Make sure you put quotes around your helper..
`````html
<img src="{{url store='originals'}}">
`````

**Comment by clarkritchie** [comment]
These are PDFs not images.  Using ```store='originals'``` causes the helper to not return anything.

**Comment by aldeed** [comment]
@clarkritchie, in your case it would be `store="files"` (whatever the store name is).

It does indeed look like we don't escape spaces [here](https://github.com/CollectionFS/Meteor-CollectionFS/blob/master/packages/access-point/access-point-common.js#L106-L111). We should be running `filename` through url escape function, so it's a bug.

The workaround for now would be to specify a spaceless filename with the helper like `{{url store='files' filename=helperToGenerateEscapedFilename}}`

**Comment by nooitaf** [comment]
I use a pdf upload thing for a while, and it has no problems with spaces. The browser magically adds the `%20` when opening the file. I remembered the quotes specialty of it though. 

Escaped or unescaped, there should be a link to click. If there is no link (no url), the problem might be somewhere else?

**Comment by aldeed** [comment]
@nooitaf, good point. We should still probably escape the url string, though, in case it's displayed somewhere on screen for copy/paste into address bar. (Even then, maybe most browsers would auto-escape it.)

**Comment by nooitaf** [comment]
@aldeed, sure. For copy/paste its better to escape, because spaces will probably not work in email, twitter or facebook.

**Comment by Shekharrajak** [comment]
This insert code is working for me : 
  
Template.upload.events({
    'id_here input[type="file"]': function(evt) {
      console.log('changed');
      FS.Utility.eachFile(evt, function(file) {
        var fileObj = new FS.File(file);
        fileObj.customProperty = 'hmmm';
        Uploads.insert(fileObj, function(err) {
          console.log(err);
        });
      });
     
    }
  });

**Comment by mikemurray** [comment]
@aldeed Yea, I think we should escape the url string, or at least the file name to avoid linking issues.
